astropy/convolution/core.py

```diff
@@ -88,42 +85,66 @@ def center(self):
         """
         return [axes_size // 2 for axes_size in self._array.shape]
 
+    def calculate_normalization(self, mode='integral'):
+        """
+        Calculate the kernel normalization factor.
+
+        Parameters
+        ----------
+        mode : {'integral', 'peak'}
+            One of the following modes:
+                * 'integral' (default)
+                    Kernel is normalized such that its integral = 1.
+                * 'peak'
+                    Kernel is normalized such that its peak = 1.
+        """
+
+        if mode == 'integral':
+            if self._array.sum() == 0:
+                self._normalization = np.inf
+            else:
+                self._normalization = 1. / self._array.sum()
+        elif mode == 'peak':
+            self._normalization = 1. / self._array.max()
+        else:
+            raise ValueError("invalid mode, must be 'integral' or 'peak'")
+        return self._normalization
+
```

Comment:
Thanks for reviewing this PR.

They are separate because `Kernel.__init__()` calls `calculate_normalization` to define `self._normalization` (but this does *not* normalize the kernel).  `self._normalization` is currently needed in `Kernel.__init__()` by `Gaussian1DKernel` and `Gaussian2DKernel` to define `self._truncation`.

The `normalize` method calls `calculate_normalization` and then also normalizes the kernel.

So, I think the options are

 1. Leave this as is.

 2. Combine `calculate_normalization` and `normalize` into a single method, remove `calculate_normalization` from `Kernel.__init__`, and in `Gaussian1DKernel` and `Gaussian2DKernel` explicitly call `self.normalize()` before defining `self._truncation`.

 3. Combine `calculate_normalization` and `normalize` into a single method, call `self.normalize()` in `Kernel.__init__`, and then remove all the separate `self.normalize()` calls in the kernel definitions (in `kernels.py`).  This would automatically normalize all kernels at initialization.  Is that desired?

Any preference?